tensorflow/examples/tutorials/word2vec/word2vec_basic.py

```diff
@@ -113,6 +124,9 @@ def generate_batch(batch_size, num_skips, skip_window):
       targets_to_avoid.append(target)
       batch[i * num_skips + j] = buffer[skip_window]
       labels[i * num_skips + j, 0] = buffer[target]
+    if len(buffer) != 0 and data_index == 0:
+      skip_to_start(buffer, skip_window)
+      continue
```

Comment:
Replace this with

    if data_index == len(data):
      buffer[:] = data[:span]
      data_index = span
    else:
      buffer.append(data[data_index])
      data_index += 1

Then you should be able to delete your `skip_to_start` function.